tests/keras/backend/test_backends.py

```diff
@@ -9,32 +9,60 @@
 from keras.utils.np_utils import convert_kernel
 
 
-def check_single_tensor_operation(function_name, input_shape, **kwargs):
-    val = np.random.random(input_shape) - 0.5
-    xth = KTH.variable(val)
-    xtf = KTF.variable(val)
+def check_single_tensor_operation(function_name, input_shape, test_input_shape=None, **kwargs):
```

Comment:
The interface to allow dynamic length in shape like `input_shape=(None, 3)` (to construct the tensor). The user can specify the actual `test_input_shape` in test time when `input_shape` has dynamic length in one dimension. see example [here](https://github.com/linxihui/keras/blob/7965fe76e65378e71b391732026b0073034ac45b/tests/keras/backend/test_backends.py#L99). The existent test function does not accept `input_shape=(None, 3)`.